lib/wasi-local-networking/src/lib.rs

```diff
@@ -139,6 +139,12 @@ impl VirtualTcpListener for LocalTcpListener {
         &mut self,
         cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<Result<usize>> {
+        {
+            let backlog = self.backlog.lock().unwrap();
+            if backlog.len() > 10 {
```

Comment:
So, if I understand correctly: here you are limiting the backlog to 10 connections.

But why are we even storing a backlog of connections?
That seems to invite all kinds of problems.

Reply:
Yeah this is because TcpListener does not support polling via the `accept_ready_poll` method so we have to simulate it... we could quite easily reduce it to a single item and use an option instead if this is nicer?

Reply:
Yeah I was also going to suggest just buffering a single one. :+1:

Reply:
We don't want to have a bunch of idle TCP connections sitting around if the thread doesn't handle them.